modules/apps/bulk/bulk-rest-impl/src/main/java/com/liferay/bulk/rest/internal/resource/v1_0/BulkActionResourceImpl.java

```diff
@@ -0,0 +1,403 @@
+import com.liferay.portal.vulcan.permission.Permission;
+
+import java.io.Serializable;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+/**
+ * @author Alejandro Tardín
+ */
+@Component(
+	properties = "OSGI-INF/liferay/rest/v1_0/bulk-action.properties",
+	scope = ServiceScope.PROTOTYPE, service = BulkActionResource.class
+)
+public class BulkActionResourceImpl extends BaseBulkActionResourceImpl {
+
+	@Override
+	public BulkActionTask postBulkAction(
+			String blueprintExternalReferenceCode, Boolean emptySearch,
+			String entryClassNames, String scope, String search, Filter filter,
+			Pagination pagination, Sort[] sorts, BulkAction bulkAction)
+		throws Exception {
+
+		if (!FeatureFlagManagerUtil.isEnabled(
+				contextCompany.getCompanyId(), "LPD-17564")) {
+
+			throw new UnsupportedOperationException();
+		}
+
+		BulkActionBulkSelectionFactory bulkActionBulkSelectionFactory =
+			_getBulkActionBulkSelectionFactory(
+				blueprintExternalReferenceCode, emptySearch, entryClassNames,
+				scope, search, filter, pagination, sorts, bulkAction);
+
+		BulkSelection<Object> bulkSelection =
+			bulkActionBulkSelectionFactory.create();
+
+		if (bulkSelection.getSize() == 0) {
+			return new BulkActionTask();
+		}
+
+		BulkAction.Type type = bulkAction.getType();
+
+		BulkActionTask bulkActionTask = _addBulkActionTask(type);
+
+		_bulkSelectionRunner.run(
+			contextUser, bulkSelection, _getBulkSelectionAction(type),
+			_getInputMap(bulkAction, bulkActionTask, type));
+
+		return bulkActionTask;
+	}
+
+	@Override
+	public Page<BulkActionItem> postBulkActionItemPreviewPage(
+			Boolean fetchChildren, String search, Filter filter,
+			Pagination pagination, Sort[] sorts, BulkAction bulkAction)
+		throws Exception {
+
+		return super.postBulkActionItemPreviewPage(
+			fetchChildren, search, filter, pagination, sorts, bulkAction);
+	}
+
+	private BulkActionTask _addBulkActionTask(BulkAction.Type type)
+		throws Exception {
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.
+				getObjectDefinitionByExternalReferenceCode(
+					"L_CMS_BULK_ACTION_TASK", contextCompany.getCompanyId());
+
+		ObjectEntry objectEntry = _objectEntryLocalService.addObjectEntry(
+			0, contextUser.getUserId(),
+			objectDefinition.getObjectDefinitionId(),
+			ObjectEntryFolderConstants.PARENT_OBJECT_ENTRY_FOLDER_ID_DEFAULT,
+			contextAcceptLanguage.getPreferredLanguageId(),
+			HashMapBuilder.<String, Serializable>put(
+				"actionName", type.toString()
+			).put(
+				"executionStatus", "initial"
+			).put(
+				"type", type.toString()
+			).build(),
+			new ServiceContext());
+
+		Map<String, Serializable> values = objectEntry.getValues();
```

Comment:
do we need to extract the values? I mean... we are using this just to extract "constants" (actionName -> type.toString(), status -> "initial", type ---> type.toString())